process/releases.md

```diff
@@ -28,3 +28,29 @@ repos. This means, that release notes have to describe:
 - changes in build requirements
 - expected build issues and possible solutions
 - known issues with workaround
+
+## Release branches
+We make the release from the main branch of our repo in most cases.
+Only in some rare cases, we release from the already existing
+platform-specific branch. But this need should be seriously considered.
+
+For the release, we should branch from the main branch with the name
+`vX.Y`. For example, if we are making release `v2.3.0` we use branch `v2.3`.
+All hotfixes have to be added to the same branch.
+So fixes for `v2.3.1` and `v.2.3.2` are merged into the same branch `v2.3`.
+For each hotfix corresponding release tag is created: `v2.3.1` and so on.
+
+In the case of the release of the moulin-based project (regular product),
+we have to use the name of the release branch as fixing revision of the
+self-referenced repo inside yaml-configuration.
+
+Example. We have repo `meta-xt-super-product` and we have to make the
+release `v2.3.0`.
+So we have to create the release branch `v2.3` and use it in the
+yaml-configuration:
+```
+    - type: git
+      url: https://github.com/xen-troops/meta-xt-super-product.git
+      rev: v2.3
```

Comment:
I am no sure I get this. Basically, you are saying that this YAML will track v2.3 branch with all its updates. How we can use this for release? It is not fixed.

Reply:
Yes. You are right. The problem is that we can't use the fixed commit or tag here, because we have to know the commit hash that doesn't exist before the merge,OR we have to use the tag that can be created only after the merge.

Reply:
I am wondering how repo-based and west.yaml-based projects tackle this issue...

Reply:
To be honest, we have one more way. To merge commit with everything fixed in recipes and in the yaml, except self-reference.  And, after that, we have to merge the commit with the self-reference pointing to the just merged commit. And to use that last commit as release one.
In this case, we will have properly working-compilable set of commits, but with one issue.
During the fetch of the sources, we will have some un-syncronization. Because we will use yaml from some release tag, and after that we will get `meta-xt-super-product` fetched to another commit that is `parent` to the release commit.
This will work, but will introduce some confusion.

Reply:
After your comment, I looked into https://github.com/xen-troops/zephyr-dom0-xt/blob/rpi5-v0.1.0/west.yml.
And I think I found the root cause and how to solve it. Our problem is already mentioned self-reference inside yaml.

Putting aside how to implement a proposed solution right now...

If we remove
```
    - type: git
      url: https://github.com/xen-troops/meta-xt-super-product.git
      rev: v2.3
```
from the `super-product.yaml`, we can merge and release properly.
During the release we just fix things, merge the commit, and put release tag on it.
When we need to build product in it's current or release state, we fetch HEAD or specific commit/tag and run moulin.
After the fetch, we have the product in the required state, and all its dependencies are described in yaml.

However, I do not know how to properly handle the situation with the fetched product that need to be placed into the expected hierarchy of build folders.

Reply:
I have removed part of the text with specific handling of the moulin-based projects, because there is no need for the special handling after the review and merge https://github.com/xen-troops/meta-xt-prod-devel-rcar/pull/143